### PR TITLE
Glutin as an optional dependency.

### PR DESCRIPTION
**Pull request by Esption**
For #675

### CODE REVIEW

**Comment by tomaka** [comment]
This requires changing the `.travis.yml` to add `glutin` in `FEATURES`.
Please also add a line with `FEATURES='' TEST=0`

**Comment by Esption** [comment]
Hope I did that correctly.  Never seen a Travis file before.

File: .travis.yml [hunk 1]
```diff
@@ -14,9 +14,10 @@ env:
     - secure: "AfE+PLANOT0mgUKpgGUu4yIrSNKxoNST5zFBeajONEEE0wl/bZVE6iGqucNPR0EaIgMfk+NGgCLLM7VFfy+HF5alBiqUUOapef+fc91iHVzoBgtlLXOO7E1UntrieBQCodURTaBtotCMe7jtlkNajkvUD6atKTYh1AYJ3JY2bHA="
     - GLIUM_HEADLESS_TESTS=1
   matrix:
+    - FEATURES='' TEST=0
     - FEATURES='headless' TEST=1
```
Comment (tomaka) [inline .travis.yml]: Here as well, or the tests won't work.

Comment (Esption) [inline .travis.yml]: Wouldn't it be better to make the tests that rely on glutin only work if glutin is enabled? I had forgotten that would be needed, but that shouldn't be too hard.

Comment (tomaka) [inline .travis.yml]: Well, yeah. 99% of the tests create a headless context and run OpenGL commands. If you remove glutin, there's nothing left to create the context.

**Comment by tomaka** [comment]
Ah, circle.yml must also be modified to pass `--features "headless glutin"` instead of just `--features "headless"`.

**Comment by tomaka** [comment]
Thanks

**Pull request merged**